Code/GraphMol/new_canon.cpp

```diff
@@ -160,20 +165,13 @@ void compareRingAtomsConcerningNumNeighbors(Canon::canon_atom *atoms,
     unsigned currentRNIdx = 0;
     atoms[idx].neighborNum.reserve(1000);
     atoms[idx].revistedNeighbors.assign(1000, 0);
-    char *visited = (char *)malloc(nAtoms * sizeof(char));
-    CHECK_INVARIANT(visited, "allocation failed");
-    memset(visited, 0, nAtoms * sizeof(char));
+
+    memset(visited.get(), 0, nAtoms * sizeof(char));
```

Comment:
Nice optimization!  Does the lack of calling malloc in the loop speed this up a bunch?

Reply:
nope, there was no significant performance impact of the changes in this PR